usersession/autostart/autostart.go

```diff
@@ -238,6 +238,20 @@ func AutostartSessionApps() error {
 
 	usrSnapDir := filepath.Join(usr.HomeDir, dirs.UserHomeSnapDir)
 
+	// restrict the user's "snap dir", i.e. /home/$USER/snap, to be private with
+	// permissions o0700 so that other users cannot read the data there, some
+	// snaps such as chromium etc may store secrets inside this directory
+	// note that this operation is safe since `userd --autostart` runs as the
+	// user so there is no issue with this modification being performed as root,
+	// and being vulnerable to symlink switching attacks, etc.
+	if err := os.Chmod(usrSnapDir, 0700); err != nil {
```

Comment:
I think this should this be more explicit, eg. in cmd_userd.go:
```
	if x.Autostart {
		// autostart is called when starting the graphical session, use that as
		// an opportunity to fix ~/snap permission bits
		if err := maybeFixupSnapPermissions(); err != nil {
			fmt.Fprintf(os.Stderr, "failed to fixup ~/snap permissions: %v\n", err)
		}
		return x.runAutostart()
	}
```

Reply:
this is done now